include/alpaka/queue/QueueUniformCudaHipRtBlocking.hpp

```diff
@@ -11,174 +12,19 @@
 
 #if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
 
-#    include <alpaka/core/Concepts.hpp>
-#    include <alpaka/dev/DevUniformCudaHipRt.hpp>
-#    include <alpaka/dev/Traits.hpp>
-#    include <alpaka/event/Traits.hpp>
-#    include <alpaka/meta/DependentFalseType.hpp>
-#    include <alpaka/queue/Traits.hpp>
 #    include <alpaka/queue/cuda_hip/QueueUniformCudaHipRtBase.hpp>
 
-// Backend specific includes.
-#    if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
-#        include <alpaka/core/Cuda.hpp>
-#    else
-#        include <alpaka/core/Hip.hpp>
-#    endif
-
-#    include <condition_variable>
-#    include <functional>
-#    include <memory>
-#    include <mutex>
-#    include <stdexcept>
-#    include <thread>
-
 namespace alpaka
 {
-    class EventUniformCudaHipRt;
-
     //! The CUDA/HIP RT blocking queue.
-    class QueueUniformCudaHipRtBlocking final : public uniform_cuda_hip::detail::QueueUniformCudaHipRtBase
-    {
-    public:
-        ALPAKA_FN_HOST QueueUniformCudaHipRtBlocking(DevUniformCudaHipRt const& dev)
-            : uniform_cuda_hip::detail::QueueUniformCudaHipRtBase(dev)
-        {
-        }
-        ALPAKA_FN_HOST auto operator==(QueueUniformCudaHipRtBlocking const& rhs) const -> bool
-        {
-            return (m_spQueueImpl == rhs.m_spQueueImpl);
-        }
-        ALPAKA_FN_HOST auto operator!=(QueueUniformCudaHipRtBlocking const& rhs) const -> bool
-        {
-            return !((*this) == rhs);
-        }
-    };
+    using QueueUniformCudaHipRtBlocking = uniform_cuda_hip::detail::QueueUniformCudaHipRtBase<true>;
```

Comment:
Do we need this file anyway now? We could just move that one line into `QueueUniformCudaHipRtBase.hpp` and delete the file. Same for the non-blocking version.

Reply:
Since it defines the user-facing type, I thought people might be using the file explicitly.
If that's not supposed to happen then, sure, I would move everything into a single file, and maybe rename it simply `QueueUniformCudaHipRt.hpp`.

Reply:
FWIW, we also follow the one-file-per-user-facing-class approach in the SYCL back-end. So this is nothing new.

Reply:
OK, I'll keep the files then.